### PR TITLE
DBZ-1340 Guard against binlog misconfiguration

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-1340

### CODE REVIEW

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java [hunk 1]
```diff
@@ -691,6 +691,9 @@ protected void handleQueryEvent(Event event) throws InterruptedException {
             // This is an XA transaction, and we currently ignore these and do nothing ...
             return;
         }
+        if (sql.toUpperCase().startsWith("INSERT ") || sql.toUpperCase().startsWith("UPDATE ") || sql.toUpperCase().startsWith("DELETE")) {
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java]: That repeated conversion in the hot path was bugging me :) I've pushed a commit which avoids it altogether.

Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java]: I knew you'll do it :-)

**Comment by jpechane** [comment]
@gunnarmorling Thanks for the improvements, applied

**Comment by devzer01** [comment]
what's the drawback on having the `binlog_image_template` set to `MINIMAL`? is it just schema validation? but arn't schema changes also pushed through binlog? and isn't that what the use of schema-registry for? trying to understand the impact. if i use an older version which doesn't enforce this requirement.

**Pull request merged**